### PR TITLE
feat: AccessLog support CEL Filter

### PR DESCRIPTION
**Pull request by zirain**
xref: #3667

### CODE REVIEW

File: internal/gatewayapi/listener.go [hunk 1]
```diff
@@ -411,3 +422,9 @@ func destinationSettingFromHostAndPort(host string, port uint32) []*ir.Destinati
 		},
 	}
 }
+
+func validCELExpression(expr string) bool {
+	env, _ := cel.NewEnv()
```
Comment (shawnh2) [inline internal/gatewayapi/listener.go:427]: is it possible to make `env` outside this func ?

File: internal/xds/translator/testdata/in/xds-ir/accesslog-cle-multi.yaml
Comment (shawnh2) [inline internal/xds/translator/testdata/in/xds-ir/accesslog-cle-multi.yaml:1]: naming: accesslog-multi-cel.yaml

**Review by shawnh2 — APPROVED** [review]
LGTM!

File: internal/gatewayapi/listener.go [hunk 2]
```diff
@@ -301,6 +302,16 @@ func (t *Translator) processAccessLog(envoyproxy *egv1a1.EnvoyProxy, resources *
 				irAccessLog.OpenTelemetry = append(irAccessLog.OpenTelemetry, al)
 			}
 		}
+
+		var validExprs []string
+		for _, expr := range accessLog.Matches {
+			if !validCELExpression(expr) {
+				t.Logger.Info("invalid CEL expression", "expression", expr)
```
Comment (arkodg) [inline internal/gatewayapi/listener.go:309]: can we collect these `errs` and update status without returning early (so we allow the good matches to be translated) ?

File: internal/gatewayapi/runner/runner.go [hunk 3]
```diff
@@ -88,6 +88,7 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 					BackendEnabled:          r.EnvoyGateway.ExtensionAPIs != nil && r.EnvoyGateway.ExtensionAPIs.EnableBackend,
 					Namespace:               r.Namespace,
 					MergeGateways:           gatewayapi.IsMergeGatewaysEnabled(resources),
+					Logger:                  r.Logger.WithName("translator"),
```
Comment (arkodg) [inline internal/gatewayapi/runner/runner.go:91]: not needed, please see above comment

File: internal/ir/xds.go [hunk 4]
```diff
@@ -1599,6 +1599,7 @@ type RateLimitValue struct {
 // AccessLog holds the access logging configuration.
 // +k8s:deepcopy-gen=true
 type AccessLog struct {
+	CELExpression []string                  `json:"celExpression,omitempty" yaml:"celExpression,omitempty"`
```
Comment (arkodg) [inline internal/ir/xds.go:1602]:
```suggestion
	CELMatches []string                  `json:"celMatches,omitempty" yaml:"celMatches,omitempty"`
```

**Review by arkodg — APPROVED** [review]
LGTM thanks !

File: internal/xds/translator/testdata/out/xds-ir/accesslog-multi-cel.listeners.yaml [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+            [%START_TIME%] "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%" %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%" "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%UPSTREAM_HOST%"
+      path: /dev/stdout
+  - filter:
+      andFilter:
+        filters:
+        - extensionFilter:
+            name: envoy.access_loggers.extension_filters.cel
+            typedConfig:
+              '@type': type.googleapis.com/envoy.extensions.access_loggers.filters.cel.v3.ExpressionFilter
+              expression: response.code >= 400
```
Comment (zhaohuabing) [inline internal/xds/translator/testdata/out/xds-ir/accesslog-multi-cel.listeners.yaml:33]: Can "response.code" or "request.url_path" be used at the listener-level logger?

Comment (zirain) [reply internal/xds/translator/testdata/out/xds-ir/accesslog-multi-cel.listeners.yaml:33]: never try that in istio, I'm not sure whether envoy know protocol at listener.

**Pull request merged**